### DISCUSSION TITLE
New Amazon Marketing Stream Demo Video 📹

### DISCUSSION THREAD

**Discussion by kdwivedi13** [amzn/ads-advanced-tools-docs#298] [Announcements]
We're excited to share our latest demo video showcasing a comprehensive reference application for Amazon Marketing Stream. This demo covers:

- Setting up Amazon Marketing Stream for Data Firehose
- Creating subscriptions
- Receiving data in an S3 bucket via Data Firehose
- Running analyses in Athena
- Visualizing analyses in Quick Sight dashboards

The best part? It's a fully serverless solution, so you only pay for what you use. By integrating with AWS Amazon Data Firehose, this reference application simplifies your workflows and reduces costs for Amazon Marketing Stream.

Check out the video now to see the demo in action! 👉 [https://www.youtube.com/watch?v=ccw0WIIAMvk&t=1391s](https://www.youtube.com/watch?v=ccw0WIIAMvk&t=1391s)

Don't miss this opportunity to level up your Amazon advertising campaigns with a streamlined and cost-effective solution. 
Get started with the reference application today.

**Comment by xinqiguo1** [comment]
When I loaded the glue tables in Athena, it only showed one record for each partition file. Could you please explain how can we transform the data to make it read for all record sizes rather than the object size? Thanks!

**Comment by kdwivedi13** [comment]
Thank you for raising this question. I found a relevant discussion that addresses this exact problem on AWS re:Post:

https://repost.aws/questions/QUnrnFmsIbTtiF8Zx-AiIQ5Q/json-table-in-glue-with-array-only-shows-first-entry

Please check this thread and let me know if it helps resolve your issue. The discussion there covers similar scenarios where Athena only shows the first entry of JSON arrays.

**Comment by xinqiguo1** [comment]
Finally figured it out! The easiest way is to enable the new line delimiter in the Firehose. Highly recommend including this part in the tutorial!
